batchflow/models/torch/benchmark.py

```diff
@@ -0,0 +1,148 @@
+    """
+    Track module #macs, #parameters, time and memory consumption on forward and backward 
+    pass for a given inputs tensor or inputs shape.
+    
+    Parameters
+    ----------
+    module : torch.nn.modules    
+    inputs : (tensor, tuple, list)
+             Data which enter to the module.
+    repeats : int
+              The number shows how many times we want to repeat
+              module for tracking memory and time.
+    warmup : int
+             Do a few iterations for stabilize std.
+    device : str
+             Device can be 'cpu' or 'gpu'.
+    track_backward : bool
+                     If True we want to track time and memory for backward operation.
+    channels_last : bool
+                    You can change strides by choosing memory format = channels_last 
+                    for your module and check how module perfomance changes.
+    amp : bool
+          Set True or False for the parameter enabled in torch.cuda.amp.autocast(enabled=amp)
+    memory_unit : str
+                  See MEMORY_UNIT_CONSTANTS which units you can choose for memory
+    
+    Returns
+    -------
+    dict
+        Keys of dict are forward/backward time/memory and values are their results.
+    """
+    
+    memory_unit_constant = MEMORY_UNIT_CONSTANTS[memory_unit]
+    
+    with TimeTracker() as total_time:
+    
+        result = {}
+
+        torch.cuda.empty_cache()
+
+        inputs = make_initialization_inputs(inputs=inputs, device=device)
+        module.to(device)
+
+        if channels_last:
+            inputs.to(memory_format=torch.channels_last)
+            module.to(memory_format=torch.channels_last)          
+
+        forward_timings = []
+        backward_timings = []
+
+        for i in range(repeats + warmup):
+
+            if i < warmup:
+                with torch.cuda.amp.autocast(enabled=amp):
+                    outputs = module(inputs)
+                outputs.backward(outputs)
+                del outputs
+                torch.cuda.empty_cache()
+                i += 1
+                continue
+                
+            with torch.cuda.amp.autocast(enabled=amp):
+                # calculate forward operation time  
+                with TimeTracker() as ft:
+                    outputs = module(inputs)
+
+                forward_time = ft.value        
+                forward_timings.append(forward_time) 
+
+            if track_backward:
+                # calculate backward operation time 
+                with TimeTracker() as bt:
+                    outputs.backward(outputs)
+                backward_time = bt.value
+                backward_timings.append(backward_time)
+
+        result['forward time mean(ms)'] = np.mean(forward_timings) 
+        result['forward time std(ms)'] = np.std(forward_timings)
+
+        # calculate forward memory
+        with MemoryTracker(device=device) as memory:
+            module(inputs)
```

Comment:
this should be 
```python
outputs = module(inputs)
```

this way, you will be able to use `outputs` in backward tracking block
Also, I'm not even sure that `cuda` performs this operation as it is not blocking